### PR TITLE
Feature/samtools

### PR DESCRIPTION
**Pull request by henrikstranneheim**

### CODE REVIEW

File: definitions/install_rd_dna_parameters.yaml [hunk 1]
```diff
@@ -110,7 +110,7 @@ eperl_5.26:
     bwakit: 0.7.15=1
     fastqc: 0.11.8=0
     sambamba: 0.6.6
-    samtools: 1.6
+    samtools: 1.9=h8ee4bcc_1
```
Comment (jemten) [inline definitions/install_rd_dna_parameters.yaml]: Might be a good idea to update samtools in the cnvnator env as well

File: lib/MIP/Program/Alignment/Samtools.pm [hunk 2]
```diff
@@ -294,284 +291,159 @@ sub samtools_index {
...
         }
     );
     return @commands;
 }
 
 sub samtools_faidx {
 
-## samtools_faidx
-
 ## Function : Perl wrapper for writing samtools faidx recipe to $FILEHANDLE. Based on samtools 1.3.1 (using htslib 1.3.1).
```
Comment (jemten) [inline lib/MIP/Program/Alignment/Samtools.pm]: Samtools 1.9?

File: lib/MIP/Program/Alignment/Samtools.pm [hunk 3]
```diff
@@ -230,40 +229,38 @@ sub samtools_view {
 
 sub samtools_index {
 
-## samtools_index
-
 ## Function : Perl wrapper for writing samtools index recipe to $FILEHANDLE. Based on samtools 1.3.1 (using htslib 1.3.1).
```
Comment (jemten) [inline lib/MIP/Program/Alignment/Samtools.pm]: Samtools 1.9?

File: lib/MIP/Recipes/Analysis/Bwa_mem.pm [hunk 4]
```diff
@@ -486,6 +486,7 @@ sub analysis_bwa_mem {
                     auto_detect_input_format => 1,
                     FILEHANDLE               => $FILEHANDLE,
                     infile_path              => $file_path,
+                    remove_overlap           => 1,
```
Comment (jemten) [inline lib/MIP/Recipes/Analysis/Bwa_mem.pm]: Version bumb? ;)

File: lib/MIP/Recipes/Install/Conda.pm [hunk 5]
```diff
@@ -135,6 +135,7 @@ sub install_conda_packages {
 ## Returns   :
 ## Arguments : $conda_env                  => Name of conda environment
 ##           : $conda_env_path             => Path to conda environment (default: conda root)
+##           : $conda_no_update_dep        => Do not update dependencies
```
Comment (jemten) [inline lib/MIP/Recipes/Install/Conda.pm]: Version bumb?

File: lib/MIP/Program/Alignment/Samtools.pm [hunk 6]
```diff
@@ -294,284 +291,159 @@ sub samtools_index {
...
 sub samtools_stats {
 
-## samtools_stats
-
-## Function : Perl wrapper for writing samtools stats recipe to $FILEHANDLE. Based on samtools 1.3.1 (using htslib 1.3.1).
-## Returns  : "@commands"
-##          : $regions_ref              => The regions to process {REF}
+## Function : Perl wrapper for writing samtools stats recipe to $FILEHANDLE. Based on samtools 1.9 (using htslib 1.9).
+## Returns  : @commands
+##          : $auto_detect_input_format => Ignored (input format is auto-detected)
```
Comment (jemten) [inline lib/MIP/Program/Alignment/Samtools.pm]: Why is it ignored?

File: definitions/install_rd_dna_parameters.yaml (hunk 1 above)
Comment (henrikstranneheim) [reply definitions/install_rd_dna_parameters.yaml]: I will wait and do that in seperate cnvnator PR

File: lib/MIP/Program/Alignment/Samtools.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Program/Alignment/Samtools.pm]: Sure I will fix it for all methods. Let's have that as a SOP to update all methods even though there are no code changes to some of them.

File: lib/MIP/Recipes/Analysis/Bwa_mem.pm (hunk 4 above)
Comment (henrikstranneheim) [reply lib/MIP/Recipes/Analysis/Bwa_mem.pm]: Oh, there is always that

File: lib/MIP/Program/Alignment/Samtools.pm (hunk 6 above)
Comment (henrikstranneheim) [reply lib/MIP/Program/Alignment/Samtools.pm]: This is just the way samtools described the option, and I think they mean that disable a check and auto-detect instead

**Pull request merged**